client/src/components/layouts/learn.tsx

```diff
@@ -86,7 +90,13 @@ function LearnLayout({
       <Helmet>
         <meta content='noindex' name='robots' />
       </Helmet>
-      <main id='learn-app-wrapper'>{children}</main>
+      <main
+        id='learn-app-wrapper'
+        {...(hasEditableBoundaries && { 'data-has-editable-boundaries': true })}
+        {...(usesMultifileEditor && { 'data-uses-multifile-editor': true })}
```

Comment:
adding `data-uses-multifile-editor` and `data-has-editable-boundaries` attributes might not needed for styling since tool panel is shown only when hasEditableBoundaries and usesMultifileEditor are True.

Reply:
I've removed `data-uses-multifile-editor` since we aren't relying on that anymore. Still need to look into `data-has-editable-boundaries`.

Reply:
Regarding `data-has-editable-boundaries`, this is used to allow us to set a different height on the tab content when the tool panel buttons aren't displayed in order to prevent double vertical scroll bars.

Reply:
That makes sense.